### PR TITLE
Fixes GH-580 Table Sort

### PR DESCRIPTION
**Pull request by mlzummo**
@GordonSmith @drealeed @jchambers-ln @jbrundage @dtsnell4 @anmoljagetia 

Please Review

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

File: src/other/Table.js [hunk 1]
```diff
@@ -235,13 +235,51 @@
...
+        // http://www.davekoelle.com/files/alphanum.js
+        // http://stackoverflow.com/questions/2802341/javascript-natural-sort-of-alphanumerical-strings
+
+        var caseSensitive = true;
+        this._data.sort(function alphanum(a, b) {
+            function chunkify(t) {
+                var tz = [];
+                var x = 0, y = -1, n = 0, i, j;
+                while (i = (j = t.charAt(x++)).charCodeAt(0)) { // jslint: suggestions?
+                    var m = (i === 46 || (i >= 48 && i <= 57));
```
Comment (mlzummo) [inline src/other/Table.js]: suppose to be == not sure how === will effect it?

Comment (jbrundage) [inline src/other/Table.js]: It shouldn't affect it at all (except maybe running slightly faster?). "===" is just a strict comparison and should behave the same as == when comparing integers.

File: src/other/Table.js [hunk 2]
```diff
@@ -235,13 +235,51 @@
...
+                return tz;
+            }
+
+            var aa = chunkify(caseSensitive ? a[idx] : a[idx].toLowerCase());
+            var bb = chunkify(caseSensitive ? b[idx] : b[idx].toLowerCase());
+
+            for (var x = 0; aa[x] && bb[x]; x++) {
+                if (aa[x] !== bb[x]) {
+                    var c = Number(aa[x]), d = Number(bb[x]);
+                    if (c == aa[x] && d == bb[x]) { // jslint: (has to be == or it doesnt work correctly) suggestions?
```
Comment (jbrundage) [inline src/other/Table.js]: Use // jshint ignore:line

File: src/other/Table.js [hunk 3]
```diff
@@ -235,13 +235,51 @@
...
-                return context._currentSortOrder;
+        // http://www.davekoelle.com/files/alphanum.js
+        // http://stackoverflow.com/questions/2802341/javascript-natural-sort-of-alphanumerical-strings
+
+        var caseSensitive = true;
+        this._data.sort(function alphanum(a, b) {
+            function chunkify(t) {
+                var tz = [];
+                var x = 0, y = -1, n = 0, i, j;
+                while (i = (j = t.charAt(x++)).charCodeAt(0)) { // jslint: suggestions?
```
Comment (jbrundage) [inline src/other/Table.js]: USe // jshint ignore:line

File: src/other/Table.js (hunk 1 above)
Comment (mlzummo) [inline src/other/Table.js]: thats not the case as we discussed earlier... just updating this comment for when gordon reviews it

**Comment by mlzummo** [comment]
branch has wrong name redoing PR

**Pull request closed without merging**